TileDB driver on windows fails to compile

```
:\PROGRA~2\MICROS~1.0\VC\bin\amd64\cl.exe  /nologo /TP -DPDAL_DLL_EXPORT=1 -DWIN32_LEAN_AND_MEAN -DWINDOWS -D_CRT_NONSTDC_NO_WARNING -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -Dlibpdal_plugin_reader_tiledb_EXPORTS -I..\vendor\gtest\include -I..\vendor\gtest -Iinclude -I..\ -I..\vendor\pdalboost -IC:\Miniconda3\envs\pdal-build\Library\include /DWIN32 /D_WINDOWS /W3 /GR /EHsc /MD /Zi /O2 /Ob1 /DNDEBUG   /wd4068 /wd4290 /wd4800 /wd4267 /wd4250 /MP8 /showIncludes /Foplugins\tiledb\CMakeFiles\libpdal_plugin_reader_tiledb.dir\io\TileDBReader.cpp.obj /Fdplugins\tiledb\CMakeFiles\libpdal_plugin_reader_tiledb.dir\ /FS -c ..\plugins\tiledb\io\TileDBReader.cpp
c:\miniconda3\envs\pdal-build\library\include\tiledb\type.h(333): warning C4003: not enough actual parameters for macro 'max'
c:\miniconda3\envs\pdal-build\library\include\tiledb\type.h(361): warning C4003: not enough actual parameters for macro 'max'
c:\miniconda3\envs\pdal-build\library\include\tiledb\type.h(361): error C2589: '(': illegal token on right side of '::'
c:\miniconda3\envs\pdal-build\library\include\tiledb\type.h(361): error C2059: syntax error: '::'
c:\miniconda3\envs\pdal-build\library\include\tiledb\type.h(361): error C2737: 'public: static unsigned int const tiledb::impl::TypeHandler<char const *,void>::tiledb_num': 'constexpr' object must be initialized
```

Typical min/max macro issues on windows, but the fixes need to be upstream @normanb

--- a/plugins/tiledb/CMakeLists.txt
+++ b/plugins/tiledb/CMakeLists.txt
@@ -19,6 +19,10 @@ target_include_directories(${reader_libname} PRIVATE
     ${PDAL_VENDOR_DIR}/pdalboost
     ${TILEDB_INCLUDE_DIR})
 
+if (WIN32)
+	target_compile_definitions(${reader_libname} PRIVATE -DNONMINMAX)
+endif()
+
 #
 # TileDB Writer
 #
@@ -33,6 +37,10 @@ PDAL_ADD_PLUGIN(writer_libname writer tiledb
 target_include_directories(${writer_libname} PRIVATE
     ${PDAL_VENDOR_DIR}/pdalboost
     ${TILEDB_INCLUDE_DIR})
+
+if (WIN32)
+	target_compile_definitions(${writer_libname} PRIVATE -DNONMINMAX)
+endif()
 
 if (WITH_TESTS)
     PDAL_ADD_TEST(pdal_io_tiledb_writer_test
@@ -50,4 +58,10 @@ if (WITH_TESTS)
             ${reader_libname}
             ${TILEDB_LIBRARIES}
     )
+
+if (WIN32)
+	target_compile_definitions(pdal_io_tiledb_writer_test PRIVATE -DNONMINMAX)
+	target_compile_definitions(pdal_io_tiledb_reader_test PRIVATE -DNONMINMAX)
+endif()
+
 endif()

--- a/plugins/tiledb/io/TileDBReader.hpp
+++ b/plugins/tiledb/io/TileDBReader.hpp
@@ -34,6 +34,8 @@
 
 #pragma once
 
+#define NOMINMAX
+
 #include <iostream>
 
 #include <pdal/Reader.hpp>

--- a/plugins/tiledb/io/TileDBWriter.hpp
+++ b/plugins/tiledb/io/TileDBWriter.hpp
@@ -34,8 +34,11 @@
 
 #pragma once
 
+#define NOMINMAX
+
 #include <pdal/Streamable.hpp>
 #include <pdal/Writer.hpp>
+
 #include <tiledb/tiledb>
 
 namespace pdal

--- a/plugins/tiledb/test/TileDBReaderTest.cpp
+++ b/plugins/tiledb/test/TileDBReaderTest.cpp
@@ -32,6 +32,8 @@
 * OF SUCH DAMAGE.
 ****************************************************************************/
 
+#define NOMINMAX
+
 #include <pdal/Filter.hpp>
 #include <pdal/pdal_test_main.hpp>
 

--- a/plugins/tiledb/test/TileDBWriterTest.cpp
+++ b/plugins/tiledb/test/TileDBWriterTest.cpp
@@ -32,9 +32,10 @@
 * OF SUCH DAMAGE.
 ****************************************************************************/
 
+#define NOMINMAX
+
 #include <stdio.h>
 #include <sys/types.h>
-#include <dirent.h>
 
 #include <pdal/pdal_test_main.hpp>
 #include <io/FauxReader.hpp>

